Dockerfile

```diff
@@ -1,57 +1,43 @@
-FROM node:20.9 AS base
-FROM node:20.9-alpine AS base-alpine
-
-# build
-FROM base AS build
-
-RUN apt-get update && apt-get install git
-
-WORKDIR /root/app
-COPY package*.json ./
-
-RUN npm ci
-
-COPY . ./
-
-RUN npx prisma generate --schema=./prisma/schema.prisma
-
-RUN npx prisma migrate
-
-RUN npm run build
-
-# development
-FROM base AS dev
-
-RUN apt-get update && apt-get install -y git \
-    postgresql-client \
-    curl \
-    vim
-
+FROM node:20.10-alpine AS base
+ENV NODE_ENV=production
+
+FROM base AS app-base
+RUN apk update \
+ && apk add tini \
+ && rm -rf /var/cache/apk/* \
+ && mkdir -p /home/node/app \
+ && chown -R node:node /home/node/app
+USER node
 WORKDIR /home/node/app
-COPY package*.json ./
-
-RUN npm install
-COPY . ./
-
-RUN npx prisma generate
-
+ENTRYPOINT [ "/sbin/tini", "--" ]
+
+FROM base AS build-base
+WORKDIR /build
+COPY package.json package-lock.json prisma ./
+RUN npm ci --include=dev --frozen-lockfile \
+ && npx prisma generate
+
+FROM build-base AS build
+COPY --chown=node:node . ./
+RUN npm run build \
+ && npm prune --production
+
+FROM app-base AS dev
+ENV NODE_ENV=development
+# used **only** for development and local directory should be mounted
 CMD [ "npm", "run", "start:debug" ]
 
-
 # production
-FROM base-alpine AS prod
-
-USER node
-
-RUN mkdir -p /home/node/app && chown -R node:node /home/node/app
-WORKDIR /home/node/app
-
-COPY --chown=node:node --from=build /root/app/package*.json ./
-COPY --chown=node:node --from=build /root/app/node_modules ./node_modules
-RUN npm prune --production
-
-COPY --chown=node:node --from=build /root/app/dist ./dist
-COPY --chown=node:node --from=build /root/app/prisma ./prisma
+FROM app-base AS prod
+COPY --chown=node:node --from=build /build/package.json /build/package-lock.json ./
+COPY --chown=node:node --from=build /build/node_modules ./node_modules
+COPY --chown=node:node --from=build /build/dist ./dist
+COPY --chown=node:node --from=build /build/prisma ./prisma
 COPY --chown=node:node nest-cli.json ./nest-cli.json
-
-ENTRYPOINT npx prisma migrate deploy && npm run start:prod
\ No newline at end of file
+EXPOSE 3000
+ARG GIT_COMMIT="undefined"
+ENV GIT_COMMIT=$GIT_COMMIT
+LABEL org.opencontainers.image.revision="${GIT_COMMIT}" \
+      org.opencontainers.image.source="https://github.com/bratislava/nest-prisma-template" \
+      org.opencontainers.image.licenses="EUPL-1.2"
+CMD [ "npm", "run", "start:migrate:prod" ]
```

Comment:
I might be overlooking something but does this command `start:migrate:prod` exists?

Reply:
thanks, i did overlooked it, now fixed